src/juju/jimm-facade.ts

```diff
@@ -16,20 +38,31 @@ class JIMMV2 {
   constructor(transport: Transport, info: ConnectionInfo) {
     this._transport = transport;
     this._info = info;
-    this.version = 2;
+    this.version = 4;
 
     // Automatically bind all methods to instances.
     autoBind(this);
   }
 
   disableControllerUUIDMasking() {
     return new Promise((resolve, reject) => {
-      const params = {};
       const req = {
         type: "JIMM",
         request: "DisableControllerUUIDMasking",
-        version: 2,
-        params: params,
+        version: 3,
+        params: {},
+      };
+      this._transport.write(req, resolve, reject);
+    });
+  }
+
+  crossModelQuery(query: string = "."): Promise<CrossModelQueryFullResponse> {
+    return new Promise((resolve, reject) => {
+      const req = {
+        type: "JIMM",
+        request: "CrossModelQuery",
+        version: 4,
+        params: { type: "jq", query },
```

Comment:
It might be worth leaving a comment to say that the API only currently supports the "jq" type so that we remember why this is hardcoded.